### PR TITLE
Exception Tasks from the Java Course

### PR DESCRIPTION
**Pull request by denisdimitrov**
The first three tasks from the Exception Topic

### CODE REVIEW

File: task3/array/Demo.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package task3.array;
+
+/**
+ * Created by clouway on 10.05.16.
+ */
+public class Demo {
+    public static void main(String[] args) throws EmptyException, FullException, PrintException {
+        Array array = new Array();
```
Comment (mgenov) [inline task3/array/Demo.java]: What will be result of:

```java
Array array = new Array();
array.addElement(8);
```

?

File: task1/sumator/NumSumator.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Created by clouway on 04.05.16.
+ */
+public class NumSumator {
+    /**
+     * Checks if the input is string or not
+     * @param a String to be checked
+     * @throws InvalidString
+     */
+    private void check(String a) throws InvalidString {
```
Comment (mgenov) [inline task1/sumator/NumSumator.java]: Private method should be placed after public methods by convention.

https://github.com/clouway/teamspace/wiki/Java%20Code%20Convention

File: task3/array/Array.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package task3.array;
+
+/**
+ * Created by clouway on 10.05.16.
+ */
+public class Array {
+    public int z = 0;
```
Comment (mgenov) [inline task3/array/Array.java]: What is the meaning of "z" ?

File: task3/array/Demo.java (hunk 1 above)
Comment (denisdimitrov) [inline task3/array/Demo.java]: Error. Because there is no array initialized.

File: task3/array/Array.java (hunk 3 above)
Comment (denisdimitrov) [inline task3/array/Array.java]: "z" is an index. I will rename it.

File: task3/array/Demo.java (hunk 1 above)
Comment (mgenov) [inline task3/array/Demo.java]: What will tell the callers that they have to call initArray method before addElement?

File: task3/array/FullException.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package task3.array;
+
+/**
+ * Created by clouway on 10.05.16.
+ */
+public class FullException extends Exception {
```
Comment (mgenov) [inline task3/array/FullException.java]: What is Full?

File: task3/array/PrintException.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package task3.array;
+
+/**
+ * Created by clouway on 10.05.16.
+ */
+public class PrintException extends Exception {
```
Comment (mgenov) [inline task3/array/PrintException.java]: What is the idea of this printing exception?

Comment (mgenov) [inline task3/array/PrintException.java]: btw, I don't see where this exception is used either.

Comment (denisdimitrov) [inline task3/array/PrintException.java]: I was not supposed to commit that. I apologize.

File: task3/array/FullException.java (hunk 4 above)
Comment (denisdimitrov) [inline task3/array/FullException.java]: The array is full. I gave the classes better names.

File: task2/rangeexception/InputScanner.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package task2.rangeexception;
+
+/**
+ * Created by clouway on 04.05.16.
+ */
+public class InputScanner {
```
Comment (mgenov) [inline task2/rangeexception/InputScanner.java]: The input comes from upper level class, so name Input doesn't sound right in this context.

Comment (denisdimitrov) [inline task2/rangeexception/InputScanner.java]: I renamed it to Range. I hope it sounds better.

Comment (mgenov) [inline task2/rangeexception/InputScanner.java]: I saw the usage now, and it doesn't look right to me. 

```java
Range range = new Range();
range.checkInput(10, 0, 100);
```

Comment (denisdimitrov) [inline task2/rangeexception/InputScanner.java]: How about now?

Comment (mgenov) [inline task2/rangeexception/InputScanner.java]: Looks better.

**Comment by mgenov** [comment]
LGTM !

**Pull request merged**